keychain_haiku.cpp

```diff
@@ -0,0 +1,129 @@
+/******************************************************************************
+ *   Copyright (C) 2011-2015 Frank Osterfeld <frank.osterfeld@gmail.com>      *
+ *                                                                            *
+ * This program is distributed in the hope that it will be useful, but        *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY *
+ * or FITNESS FOR A PARTICULAR PURPOSE. For licensing and distribution        *
+ * details, check the accompanying file 'COPYING'.                            *
+ *****************************************************************************/
+#include "keychain_p.h"
+
+#include <KeyStore.h>
+#include <QDebug>
+
+using namespace QKeychain;
+
+static QString strForStatus( status_t os ) {
+    const char * const buf = strerror(os) ;
+    return QObject::tr( "error 0x%1: %2" )
+            .arg( os, 8, 16 ).arg( QString::fromUtf8( buf, strlen( buf ) ) );
+}
+
+void ReadPasswordJobPrivate::scheduledStart()
+{
+    QString errorString;
+    Error error = NoError;
+    BKeyStore keyStore;
+    BPasswordKey password;
+
+    status_t result = keyStore.GetKey(B_KEY_TYPE_PASSWORD,
+                                      q->service().toUtf8().constData(),
+                                      q->key().toUtf8().constData(),
+                                      false, password);
+
+    data = QByteArray((char*)password.Data(), password.DataLength());
```

Comment:
Don't use C casts. use `static_cast<>`  (or `reinterpret_cast<>`)

Reply:
Ok, I copy-pasted this line from the win code :D